### PR TITLE
[core] Polylabel-based "pole of inaccessibility" symbol placement

### LINKED ISSUE #6115: Add support for pole-of-inaccessibility symbol placement
GL JS now supports pole-of-inaccessibility symbol placement for polygon geometries. We need to port this feature over to GL Native. 
- https://github.com/mapbox/mapbox-gl-js/issues/1231
- https://github.com/mapbox/mapbox-gl-js/pull/2678
- https://github.com/mapbox/mapbox-gl-test-suite/pull/111

cc @blanchg @jfirebaugh @mourner

### PR DESCRIPTION
**Pull request by jfirebaugh**
Fixes #6115

### CODE REVIEW

File: src/mbgl/layout/symbol_layout.cpp [hunk 1]
```diff
@@ -301,12 +303,20 @@ void SymbolLayout::addFeature(const SymbolFeature& feature,
...
+            Polygon<double> poly;
+            for (const auto& ring : polygon) {
+                LinearRing<double> r;
+                for (const auto& p : ring) {
+                    r.push_back(convertPoint<double>(p));
+                }
+                poly.push_back(r);
             }
+
+            // 16 here represents 2 pixels
```
Comment (kkaefer) [inline src/mbgl/layout/symbol_layout.cpp]: why?

Comment (mourner) [inline src/mbgl/layout/symbol_layout.cpp]: Yes, exactly.

Comment (kkaefer) [inline src/mbgl/layout/symbol_layout.cpp]: Ok; can we replace this with the symbolic constants that we're using throughout GL native?

Comment (jfirebaugh) [inline src/mbgl/layout/symbol_layout.cpp]: Doesn't 16 equal 1 pixel though? 8192/512=16

Comment (blanchg) [inline src/mbgl/layout/symbol_layout.cpp]: That parameter is the accuracy threshold, I.e. stop searching if the target size reaches 16 or less.  @mourner helped me to identify that. Is it possible at the time we were working off of 256 tiles?

**Pull request merged**